### ISSUE TITLE
[REVIEW]: Minerva: a light-weight, narrative image browser for multiplexed tissue images

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2579]
**Submitting author:** @thejohnhoffer (<a href="http://orcid.org/0000-0002-5877-4266">John Hoffer</a>)
**Repository:** <a href="https://github.com/labsyspharm/minerva-story" target ="_blank">https://github.com/labsyspharm/minerva-story</a>
**Version:** 1.0.0
**Editor:** @jni
**Reviewer:** @will-moore, @sofroniewn
**Archive:** <a href="https://doi.org/10.5281/zenodo.4088741" target="_blank">10.5281/zenodo.4088741</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/4b67e6fbb2e3ffe625edd3b6e6ce767f/status.svg)](https://joss.theoj.org/papers/4b67e6fbb2e3ffe625edd3b6e6ce767f)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/4b67e6fbb2e3ffe625edd3b6e6ce767f"><img src="https://joss.theoj.org/papers/4b67e6fbb2e3ffe625edd3b6e6ce767f/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/4b67e6fbb2e3ffe625edd3b6e6ce767f/status.svg)](https://joss.theoj.org/papers/4b67e6fbb2e3ffe625edd3b6e6ce767f)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/labsyspharm/minerva-story" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@will-moore & @sofroniewn, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @jni know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/labsyspharm/minerva-story">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@thejohnhoffer) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/labsyspharm/minerva-story">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@thejohnhoffer) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @will-moore, @sofroniewn it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1016/j.cels.2016.03.008 is OK
- 10.1016/j.cell.2020.03.053 is OK
- 10.1051/0004-6361/201322068 is OK
- 10.1158/2159-8290.CD-12-0095 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02579/joss.02579/10.21105.joss.02579.pdf)

**Comment by jni** [comment]
btw @sofroniewn @will-moore feel free to ping me with any questions about the process!

**Comment by will-moore** [comment]
I'm failing at "Is the source code for this software available at the repository url?" for 2 reasons:
 - Seems that all the JS code is minified in https://github.com/labsyspharm/minerva-story/blob/master/_includes/bundle.js and the MinervaStory source code isn't included in the repository
 - Also, half the paper is about https://github.com/labsyspharm/minerva-author so should the checklist above be duplicated for that repo?

**Comment by jni** [comment]
I think this means @thejohnhoffer needs to update the code in the repo to be readable, and include the code for minerva-author somewhere in the submission repo as well, but I hope @lpantano can confirm this or clarify...

**Comment by jni** [comment]
Just got confirmation from @arfon:

- The authors should unminify all code for review, though that bundle might be external libraries?

- The source code under review should preferably be together in this repo, but otherwise, the authors should point to the source code to be reviewed, *and* they will have to produce a complete bundle for the Zenodo archive at the time of acceptance.

**Comment by jni** [comment]
@thejohnhoffer Did you see the comments above? I think at this point the reviewers are waiting on you to clarify where the code under review is.

**Comment by thejohnhoffer** [comment]
Hi @jni and @will-moore .

I'm sorry for the confusion. We have indeed split the code for this project among four separate repositories. Instructions for where to access these repositories is hidden on the [Minerva Story Wiki](https://github.com/labsyspharm/minerva-story/wiki/For-Developers), but I'll replicate that here.

### Minerva Story
The GitHub Pages site build is stored at [minerva-story](https://github.com/labsyspharm/minerva-story). The source code for the minified bundle is stored at [minerva-browser](https://github.com/labsyspharm/minerva-browser).

### Minerva Author
The Python Flask server along with automated testing is stored at [minerva-author](https://github.com/labsyspharm/minerva-author). The React UI is stored at [minerva-author-ui](https://github.com/labsyspharm/minerva-author-ui)

Again, I do apologize. Minerva Story is the repository to host the final product in the form for a Jekyll blog, but Minerva as a whole contains much more informative code in the other three repositories.

**Comment by will-moore** [comment]
Hi @thejohnhoffer, thanks for that.
Is there any more info on how to build the source code from https://github.com/labsyspharm/minerva-browser into the bundle.js at https://github.com/labsyspharm/minerva-story/blob/master/_includes/bundle.js? 
I can see a webpack build config etc for minerva-author-ui but not for minerva-browser? 
 Many thanks,
  Will.

**Comment by thejohnhoffer** [comment]
Hi @will-moore ,

We have a webpack build config, but it has not yet been integrated into either minerva-story or minerva-browser. Today, I will refactor the code slightly to include in the minerva-story repo a proper build setup for the minerva-browser bundle.

**Comment by thejohnhoffer** [comment]
Hi @will-moore,

I've updated Minerva Story [to include webpack instructions](https://github.com/labsyspharm/minerva-story#rebuilding-minerva-browser) for rebuilding Minerva Browser.

**Comment by will-moore** [comment]
I've checked off all the items that I feel able to just now.
I have issues setting up ruby on my Mac to test the install instructions, but I'll continue to look at that.
Other issues are:
 - I'm not sure if API documentation is essential, given that this is presented as a GUI tool rather than a software library?
 - I couldn't see any Automated tests or community guidelines for contributing.
 - The paper doesn't compare other tools for viewing large multiplexed images on the web.

**Comment by sofroniewn** [comment]
I will start my review asap @jni, thanks @will-moore for ironing out some of those early issues!

**Comment by jni** [comment]
Just a status report from my perspective:

- @thejohnhoffer, could you address the [comments](https://github.com/openjournals/joss-reviews/issues/2579#issuecomment-684823383) from @will-moore above?
- We are still waiting for review from @sofroniewn, but he has already noted that it's on the way.

**Comment by jmuhlich** [comment]
Regarding the comparison to other tools, we thought the reference to our other publication Rashid 2020 (in review at another journal) would be sufficient -- that work surveys the field much more comprehensively than we have room for here. Would you prefer we include a small summary of that review in this manuscript? If not, we can at least make it more clear in the text that this information can be found in that reference.

We'll certainly add community guidelines and think about what we can create automated tests for.

**Comment by will-moore** [comment]
Hi @jmuhlich, Thanks, I see that Rashid et.al exactly ticks the box of "describe how this software compares to other commonly-used packages" but I guess that could be made a little clearer. 
I found that the DOI for Rashid seems to be incorrect: https://doi.org/10.1051/0004-6361/201322068
Google led me to https://www.biorxiv.org/content/10.1101/2020.03.27.001834v1 but I ended up reading the current version at https://www.biorxiv.org/content/10.1101/2020.03.27.001834v2.
Thanks,

**Comment by thejohnhoffer** [comment]
I've updated the paper to mention that Rashid 2020 contains a comparison of existing software, I've also updated the DOI for that particular paper. We've also added [contribution guidelines to the wiki](https://github.com/labsyspharm/minerva-story/wiki/Contribution-Guidelines). Also, I'd like to point out that we have tests for Minerva Author [here](https://github.com/labsyspharm/minerva-author/blob/master/src/test_app.py) and [here](https://github.com/labsyspharm/minerva-author/blob/master/src/test_render_jpg.py) that can be run with a `pytest` command.

@will-moore are you still having trouble install Ruby on MacOS? On our wiki, we have two options for installing [Ruby on MacOS](https://github.com/labsyspharm/minerva-story/wiki/How-to-run-Minerva-Story-locally%3F#installing-ruby-on-macos). The first option uses RVM, which can avoid a lot of the issues on MacOS with the system version of Ruby. If you're still stuck, would you be able to share the error you're seeing?

**Comment by will-moore** [comment]
Thanks @thejohnhoffer. The install for minerva-story and minerva-author worked fine for me. The only step missing from https://github.com/labsyspharm/minerva-author#installing that I had to do was `brew install openslide`.
I was able to run the example from minerva-story.
For minerva-author I can start the app but currently I never get further than `Opening file: /path/to/image.tiff`.
This seems to hang for me.
But don't think that needs to block any of my checklist, so I think I'm done.

**Comment by jni** [comment]
@whedon re-invite @sofroniewn as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@sofroniewn please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by sofroniewn** [comment]
Thanks @jni, permissions updated, I've begun the review process, including making some minor suggestions here https://github.com/labsyspharm/minerva-story/issues/63. Will press on with further review promptly

**Comment by jni** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02579/joss.02579/10.21105.joss.02579.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02579/joss.02579/10.21105.joss.02579.pdf) :page_facing_up: :point_left:

**Comment by sofroniewn** [comment]
I've gone through the software paper and am happy with it, I think the added information that Rashid 2020 contains a comparison of existing software is sufficient to detail the state of the field.

I will continue to review the documentation and the functionality.

**Comment by sofroniewn** [comment]
I made an issue asking for the documentation links to be added to the readme so they are easier to discover.

I've gone through the documentation and functionality and found them to be satisfactory.

I found [contributing guidelines](https://github.com/labsyspharm/minerva-story/wiki/Contribution-Guidelines) here.

I wasn't able to find a full API reference say for code in https://github.com/labsyspharm/minerva-browser, but I think given the nature of this project as a web app, I don't think the target audience or potential contributors are expecting a full API reference and the code does contain helpful comments and appears clearly written, so I think it is sufficiently documented.

I wasn't able to find any continuous integration or automated tests, so I have left that checkbox blank, but I think that is not unusual for web projects of this nature and so I don't have any additional concerns.

I really like the examples present in the `featured stories` on [the main software website](https://www.cycif.org/software/minerva). They are a great way to get a sense for what can be built with the tool and a highlight of the submission.

I have not made my own story, but I was able to install and successfully install and launch the software and am convinced by the high quality examples (which I have run in my browser) that the functionality exists as described in the manuscript and the performance of the software (interactive panning/ zooming of large datasets) matches the expectations described in the manuscript.

I now consider my review complete and approve this submission, but I'm happy to expand on any of my statements or look into things more at the request of @jni.

Great job authors, congrats on a really nice piece of software and manuscript. I'm excited to see the community pick this up!!

**Comment by jni** [comment]
Thank you @will-moore and @sofroniewn for your reviews!

@will-moore can you confirm that you're happy to accept the manuscript as it stands?

Thanks everyone for the interaction to improve the paper and the software thus far!

**Comment by thejohnhoffer** [comment]
I'd like to mention that to resolve [issues with ruby dependencies](https://github.com/labsyspharm/minerva-story/issues/63), we have updated the manuscript and the Minerva Story repository to no longer require the use of Jekyll (or Ruby).

**Comment by jmuhlich** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02579/joss.02579/10.21105.joss.02579.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02579/joss.02579/10.21105.joss.02579.pdf) :page_facing_up: :point_left:

**Comment by jni** [comment]
@whedon accept

**Comment by whedon** [comment]
No archive DOI set. Exiting...

**Comment by jni** [comment]
LOL This is my first time accepting a JOSS paper so bear with me... 😅

**Comment by jni** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.cels.2016.03.008 is OK
- 10.1016/j.cell.2020.03.053 is OK
- 10.1101/2020.03.27.001834 is OK
- 10.1158/2159-8290.CD-12-0095 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by jni** [comment]
@thejohnhoffer @jmuhlich I checked the final proof for typos/small errors, and found just two very minor issues:

- the title for Rashid R et al is "Interpretative guides for interacting with tissue atlas and digital pathology data using the minerva browser" in the paper, but "Online narrative guides for illuminating tissue atlas data and digital pathology images" on biorxiv.
- "targetting" should be changed to "targeting" (according to [this page](https://english.stackexchange.com/questions/205815/is-targetted-a-standard-british-english-spelling)).

After those small issues are fixed, we need you to make a tagged release and archive of the software, and to report the version number and DOI here. (That was Whedon's issue above 😂.)

**Comment by thejohnhoffer** [comment]
@jni I've fixed those two issues with the paper. Thanks for finding them. For the release and archive of our software, do you have any suggestions as to how we should handle our four separately versioned repositories? Also, for the archive process, my current idea is to [follow GitHub's guide for creating citable archives.](https://guides.github.com/activities/citable-code/)

**Comment by jni** [comment]
@thejohnhoffer great questions! To me it sounds good but I hope @lpantano and/or @arfon can provide a more definitive answer.

**Comment by lpantano** [comment]
I think that sounds good, I will let @arfon to re-confirm. As far as there is an isolated download from Zenodo that makes this reproducible to this point, it should be good (i think).

**Comment by arfon** [comment]
> For the archive process, our current idea is to use Zenodo to archive a zip file containing exports of our four separate repositories. We would version releases for our repositories separately, and we'd have a README in the Zenodo archive pointing to the four separate releases on github. The paper itself would be within the minerva-story folder of the Zenodo archive. Would this be an okay way to handle an archive for our four repositories?

Sounds great. Thanks for confirming @thejohnhoffer.

**Comment by thejohnhoffer** [comment]
@jni @lpantano @arfon,

Thank you all. I've archived releases for all four repositories in [one archive on on Zenodo](https://zenodo.org/record/4088741#.X4dVAxNKjOQ).

**Comment by jni** [comment]
@whedon set 10.5281/zenodo.4088741 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4088741" target="_blank">10.5281/zenodo.4088741</a> is the archive.

**Comment by jni** [comment]
@whedon set 1.0.0 as version

**Comment by whedon** [comment]
OK. 1.0.0 is the version.

**Comment by jni** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.cels.2016.03.008 is OK
- 10.1016/j.cell.2020.03.053 is OK
- 10.1101/2020.03.27.001834 is OK
- 10.1158/2159-8290.CD-12-0095 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1810

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1810, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@thejohnhoffer  - can you please merge in the changes in https://github.com/labsyspharm/minerva-story/pull/66 ?

**Comment by jmuhlich** [comment]
@danielskatz Merged, thanks.

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.cels.2016.03.008 is OK
- 10.1016/j.cell.2020.03.053 is OK
- 10.1101/2020.03.27.001834 is OK
- 10.1158/2159-8290.CD-12-0095 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1813

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1813, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1316733638617313280) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1814
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02579](https://doi.org/10.21105/joss.02579)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by danielskatz** [comment]
Congratulations to @thejohnhoffer (John Hoffer) and co-authors!!

And thanks to @will-moore & @sofroniewn for reviewing, and @jni for editing!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02579/status.svg)](https://doi.org/10.21105/joss.02579)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02579">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02579/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02579/status.svg
   :target: https://doi.org/10.21105/joss.02579
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02579/status.svg)](https://doi.org/10.21105/joss.02579)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
